### PR TITLE
fix: crash when there are no bdevs and lvm export

### PR DESCRIPTION
**Pull request by tiagolobocastro**
fix: crash when there are no bdevs
    
    When there are no bdevs the iterator will return nothing.
    In this case we should not crash.
    
    Signed-off-by: Tiago Castro <tiagolobocastro@gmail.com>

---

    fix(lvm): don't export if not enabled
    
    There's no point trying and erroring with message..
    
    Signed-off-by: Tiago Castro <tiagolobocastro@gmail.com>

### CODE REVIEW

**Review by chriswldenyer — APPROVED** [review]
Looks reasonable

File: io-engine/src/grpc/v1/snapshot.rs
Comment (dsharma-dc) [inline io-engine/src/grpc/v1/snapshot.rs:1]: The change in this file is unrelated to the crash w.r.t lvm. You may want to split the commit.

**Review by dsharma-dc — APPROVED** [review]

File: io-engine/src/grpc/v1/snapshot.rs
Comment (tiagolobocastro) [reply io-engine/src/grpc/v1/snapshot.rs:1]: Indeed, done

**Comment by tiagolobocastro** [comment]
bors merge
